tests/apollo/test_skvbc_dbsnapshot.py

```diff
@@ -292,14 +291,68 @@ async def test_get_dbcheckpoint_info_request_cmd(self, bft_network, tracker):
+        """
+        bft_network.start_all_replicas()
+        client = bft_network.random_client()
+        skvbc = kvbc.SimpleKVBCProtocol(bft_network, tracker)
+        for i in range(300):
+            await skvbc.send_write_kv_set()
+        # There will be 2 dbcheckpoints created on stable seq num 150 and 300.
+        await self.wait_for_stable_checkpoint(bft_network, bft_network.all_replicas(), 300)
+        await self.wait_for_created_snapshots_metric(bft_network, 2)
+
+        op = operator.Operator(bft_network.config, client, bft_network.builddir)
+        rep = await op.get_dbcheckpoint_info_request(False)
+        data = cmf_msgs.ReconfigurationResponse.deserialize(rep)[0]
+        self.assertTrue(data.success)
+        for r in client.get_rsi_replies().values():
+            res = cmf_msgs.ReconfigurationResponse.deserialize(r)
+            self.assertEqual(len(res[0].response.db_checkpoint_info), 2)
+            dbcheckpoint_info_list = res[0].response.db_checkpoint_info
+            self.assertTrue(any(dbcheckpoint_info.seq_num == 150 for dbcheckpoint_info in dbcheckpoint_info_list))
+            self.assertTrue(any(dbcheckpoint_info.seq_num == 300 for dbcheckpoint_info in dbcheckpoint_info_list))
+        last_blockId =  await bft_network.get_metric(0, bft_network, "Gauges", "lastDbCheckpointBlockId", component="rocksdbCheckpoint")
+        for replica_id in range(len(bft_network.all_replicas())):
+            self.verify_snapshot_is_available(bft_network, replica_id, last_blockId)
+
+        # Now, crash one of the non-primary replicas
+        crashed_replica = 1
+        bft_network.stop_replica(crashed_replica)
+        # Make sure the system is able to make progress
+        for i in range(150):
+            await skvbc.send_write_kv_set()
+        await self.wait_for_stable_checkpoint(bft_network, bft_network.all_replicas(without={crashed_replica}), 450)
+
+        op = operator.Operator(bft_network.config, client, bft_network.builddir)
+        rep = await op.get_dbcheckpoint_info_request(True)
+        data = cmf_msgs.ReconfigurationResponse.deserialize(rep)[0]
+        assert(data.success == True)
+        for r in client.get_rsi_replies().values():
+            res = cmf_msgs.ReconfigurationResponse.deserialize(r)
+            dbcheckpoint_info_list = res[0].response.db_checkpoint_info
+            self.assertEqual(len(res[0].response.db_checkpoint_info), 3)
+            assert(any(dbcheckpoint_info.seq_num == 150 for dbcheckpoint_info in dbcheckpoint_info_list))
+            assert(any(dbcheckpoint_info.seq_num == 300 for dbcheckpoint_info in dbcheckpoint_info_list))
+            assert(any(dbcheckpoint_info.seq_num == 450 for dbcheckpoint_info in dbcheckpoint_info_list))
+        last_blockId =  await bft_network.get_metric(0, bft_network, "Gauges", "lastDbCheckpointBlockId", component="rocksdbCheckpoint")
+        for replica_id in range(len(bft_network.all_replicas())):
```

Comment:
we can do this:
live_replicas =  bft_network.all_replicas(without=set(crashed_replica))
for replica_id in live_replicas:
    last_blockId = await bft_network.get_metric(replica_id, bft_network, "Gauges", "lastDbCheckpointBlockId", component="rocksdbCheckpoint")
    self.verify_snapshot_is_available(bft_network, replica_id, last_blockId)

Reply:
Fixed.